### PR TITLE
Tools #2.

### PR DESCRIPTION
**Pull request by kali-03**

### CODE REVIEW

File: functions.cpp [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#include "functions.hpp"
+#include <iostream>
+#include <algorithm>
+
+void print(std::vector<int> const & numbers) {
+    for (int i = 0; i < numbers.size(); ++i) {
```
Comment (Pawellos) [inline functions.cpp]: I think you should use size_t unless int

Comment (Pawellos) [reply functions.cpp]: I supposed if you would add more flags it will be warning/error

File: Makefile [hunk 2]
```diff
@@ -1,3 +1,3 @@
 all: src/*.cpp inc/*.hpp
-	g++ -std=c++17 src/*.cpp -o main
+	g++ -std=c++17 src/*.cpp -Werror -I./src -I./inc -o main
```
Comment (Pawellos) [inline Makefile]: You could add more flags. Like -Wall -Wextra -pedantic -Wconversion

File: functions.cpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#include "functions.hpp"
+#include <iostream>
+#include <algorithm>
```
Comment (Marcinati) [inline functions.cpp]: Sort headers.

File: src/functions.cpp [hunk 4]
```diff
@@ -1,10 +1,10 @@
-#include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
+#include "functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
```
Comment (Marcinati) [inline src/functions.cpp]: Delete comment.

File: src/main.cpp [hunk 5]
```diff
@@ -1,6 +1,6 @@
 #include <vector>
 #include <iostream>
-#include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
+#include "functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
```
Comment (Marcinati) [inline src/main.cpp]: Sort headers & delete comment.

**Review by Kasperice — CHANGES_REQUESTED** [review]
As I remember file "functions.cpp" should not be included in solution. Its content should be moved to `src/functions.cpp`

File: src/main.cpp [hunk 6]
```diff
@@ -10,5 +10,6 @@ int main() {
     auto minimum = min(numbers);
     std::cout << "Minimum: " << minimum << '\n';
     
+
```
Comment (ziobron) [inline src/main.cpp]: Where is reverse function? It's gone. It wasn't merged properly. We want both min and reverse functions.

**Review by ziobron — CHANGES_REQUESTED** [review]

File: src/functions.cpp [hunk 7]
```diff
@@ -1,10 +1,10 @@
-#include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
 #include <algorithm>
+#include "functions.hpp"
```
Comment (DamiXioR) [inline src/functions.cpp:2]: you should move include header on the top

**Review by DamiXioR — COMMENTED** [review]
Good job

**Comment by ziobron** [comment]
+4 XP Narzędzia#2 – vimtutor
+5 XP Narzędzia#2 – git - konflikty
+4 XP Narzędzia#2 – flagi kompilacji
+3 XP Narzędzia#2 – git - cherrypick
+8 XP delivery before 07.06.2020

{ "requirement_ids": [1119, 1999, 1119, 2000, 1119, 2001, 1119, 2002], "points": [2, 4, 2, 5, 2, 4, 2, 3] }

**Pull request closed without merging**